File: typical90/007/07.py
from bisect import *
n = int(input())
a = list(map(int, input().split()))
a.sort()
for _ in range(int(input())):
	b = int(input())
	le = bisect_left(a, b)
	ans = 1 << 60
	if le > 0:
		ans = min(ans, abs(a[le - 1] - b))
	if le < n:
		ans = min(ans, abs(a[le] - b))
	print(ans)

File: typical90/007/07.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N;
	cin >> N;
	vector<int> A(N);
	for (int i = 0; i < N; i++) {
		cin >> A[i];
	}
	int Q;
	cin >> Q;
	vector<int> B(Q);
	for (int i = 0; i < Q; i++) {
		cin >> B[i];
	}

	sort(A.begin(), A.end());
	for (int i: B) {
		int x = lower_bound(A.begin(), A.end(), i) - A.begin();
		int ans = 1e9;
		if (x > 0) {
			ans = min(ans, i - A[x - 1]);
		}
		if (x < N) {
			ans = min(ans, A[x] - i);
		}
		cout << ans << '\n';
	}
}